mining/worker.go

```diff
@@ -166,65 +160,83 @@ func (w *DefaultWorker) Mine(ctx context.Context, base types.TipSet, nullBlkCoun
 		return false
 	}
 
-	st, err := w.getStateTree(ctx, base)
-	if err != nil {
-		log.Errorf("Worker.Mine couldn't get state tree for tipset: %s", err.Error())
-		outCh <- Output{Err: err}
-		return false
-	}
-
 	log.Debugf("Mining on tipset: %s, with %d null blocks.", base.String(), nullBlkCount)
 	if ctx.Err() != nil {
 		log.Warningf("Worker.Mine returning with ctx error %s", ctx.Err().Error())
 		return false
 	}
 
-	challenge, err := consensus.CreateChallengeSeed(base, uint64(nullBlkCount))
+	// Read uncached worker address
+	workerAddr, err := w.api.MinerGetWorkerAddress(ctx, w.minerAddr)
```

Comment:
This is one of those calls that should be clearly parameterised by the tipset keying the state in which the address is looked up, #3022. Can you make that change in a prefactor?

Reply:
#3401 addresses this.  It may merge first, but I think this can go in before that if it is ready to go first as the race #3401 prevents is rare and is unlikely to cause test flakiness / bad user behavior in a short amount of time.